### PR TITLE
Francescalb/testing dependencies

### PR DESCRIPTION
**Pull request by francescalb**
# Description:

## Type of change:

- [ ] Bug fix.
- [ ] New feature.
- [ ] Documentation update.
- [ ] Testing.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: tests/test_add_function.py [hunk 1]
```diff
@@ -1,68 +1,75 @@
...
-    standard="fno",
-)
-assert (
-    ts.serialize().strip()
-    == """
+    def func(a, b):
+        """Returns the sum of `a` and `b`."""
+        return a + b
+
+    f_id = function_id(func)
```
Comment (jesper-friis) [inline tests/test_add_function.py:17]: Very nice idea to store the f_id such that it can be reused in the assertments below :-)

**Review by jesper-friis — APPROVED** [review]
Given that we want to tie ourselves to pytest this is a great PR that does what it intends to do and nothing more. Approves under that premises.

The bad thing is that the tests becomes less readable (or at least it require that the reader spends time that could have been used for something else to learn pytest) and makes interactive debugging with e.g. ipython impossible (because the pytest --pdb option is broken).

Another possibility would be to find an alternative test runner that doesn't load the tests without properly importing them and fails for missing packages even for test scripts that exit gracefully if depending packages are not installed.

File: tests/test_add_function.py (hunk 1 above)
Comment (CasperWA) [reply tests/test_add_function.py:17]: Thanks :) That was my idea

File: tripper/mappings/mappings.py [hunk 2]
```diff
@@ -9,15 +9,14 @@
 with numpy.
 
 """
-# pylint: disable=invalid-name,redefined-builtin
+# pylint: disable=invalid-name,redefined-builtin, too-many-lines
```
Comment (CasperWA) [inline tripper/mappings/mappings.py:12]:
```suggestion
# pylint: disable=invalid-name,redefined-builtin,too-many-lines
```

File: tripper/mappings/mappings.py [hunk 3]
```diff
@@ -9,15 +9,14 @@
...
 from __future__ import annotations  # Support Python 3.7 (PEP 585)
 
 import subprocess  # nosec: B404
 from collections import defaultdict
 from enum import Enum
 from typing import TYPE_CHECKING, Mapping, Sequence
 
-import numpy as np
 from pint import Quantity  # remove
```
Comment (CasperWA) [inline tripper/mappings/mappings.py:20]: This is not part of this PR, so this comment can be disregarded (or a follow-up issue opened).

Why is there a `# remove` comment for this line?

File: tripper/utils.py [hunk 4]
```diff
@@ -39,6 +39,12 @@ class UnusedArgumentWarning(Warning):
 
 def infer_iri(obj):
     """Return IRI of the individual that stands for object `obj`."""
+
+    # Please note that tripper does not depend on neither DLite nor Pydantic.
+    # Hence neither of these packages are imported.  However, due to duck-
+    # typing, infer_iri() is still able to recognise DLite and Pydantic
+    # objects and infer their IRIs.
```
Comment (CasperWA) [inline tripper/utils.py:46]: Good comment 👍

**Review by CasperWA — CHANGES_REQUESTED** [review]
The only "hard" change I'd request is the usage of v4 for the `actions/checkout` GitHub action in `ci_tests.yml`. As well as possibly adding version specifiers for `pytest` and `pytest-cov` in the same file.

Excellent work here 👍

File: tripper/mappings/mappings.py (hunk 3 above)
Comment (francescalb) [reply tripper/mappings/mappings.py:20]: Don't know. I'd leave it for now.

File: tests/convert/test_convert.py [hunk 5]
```diff
@@ -1,55 +1,60 @@
...
-d1b = load_container(ts, EX.config1b, recognised_keys="basic")
-d2b = load_container(ts, EX.config2b, recognised_keys="basic")
-
-assert d1b == config1
-assert d2b == config2
+
+
+def test_convertions():
+    """Test convertions. Uses rdflib as triplestore backend"""
+    import pytest
```
[the remark this answers is not in the document]
Comment (francescalb) [reply tests/convert/test_convert.py:7]: I have thought about it. I really prefer to do the importorskip for each test-function. To make sure that one does not accidentally skip new function added at a later stage. It is easy to miss out on things at the top of the file. 

pytest can of course be moved outside.

File: tests/test_namespace.py [hunk 6]
```diff
@@ -17,6 +17,8 @@ def test_namespaces(get_ontology_path: "Callable[[str], Path]") -> None:
             `pathlib.Path` object pointing to an ontology test file.
 
     """
+
+    pytest.importorskip("rdflib")
```
[the remark this answers is not in the document]
Comment (francescalb) [reply tests/test_namespace.py:21]: I prefer to keep it inside, as explained above.

File: .github/workflows/ci_tests.yml [hunk 7]
```diff
@@ -85,3 +85,32 @@ jobs:
...
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        sudo apt install graphviz
+        python -m pip install -U pip
+        pip install -U setuptools wheel flit
+        pip install pytest
+        pip install pytest-cov
```
[the remark this answers is not in the document]
Comment (francescalb) [reply .github/workflows/ci_tests.yml:112]: I prefer option 2

**Review by CasperWA — APPROVED** [review]
All good 👍 

Please open follow-up issues according to the comments in my previous review.

As a tip, these can easily be created in a "quick-and-dirty"-style from a given comment: 
image

File: tests/test_get_data.py [hunk 8]
```diff
@@ -1,77 +1,77 @@
...
+        return Value([1, 7, 2, 3], unit="m")
+
+    def get_string(iri, configurations, triplestore):
+        """Return data from a data source. Here a string."""
+        return "a string"
+
+    # def get_number(iri, configurations, triplestore):
+    #    """Return data from a data source. Here a number."""
+    #    return 2.78
+
```
[the remark this answers is not in the document]
Comment (francescalb) [reply tests/test_get_data.py:25]: No, I'll keep this and make an issue.

File: tripper/mappings/mappings.py (hunk 3 above)
Comment (francescalb) [reply tripper/mappings/mappings.py:20]: made issue

File: tests/test_add_function.py [hunk 9]
```diff
@@ -1,68 +1,75 @@
 """Test Triplestore.add_function()"""
 # pylint: disable=invalid-name
-from tripper import Triplestore
 
 
-def func(a, b):
-    """Returns the sum of `a` and `b`."""
-    return a + b
+def test_add_function():
```
[the remark this answers is not in the document]
Comment (francescalb) [reply tests/test_add_function.py:5]: made issue

**Pull request merged**